crates/core/tedge/src/cli/init.rs

```diff
@@ -57,140 +62,43 @@ impl TEdgeInitCmd {
                 Some(self.user.clone()),
                 Some(self.group.clone()),
-                Some(0o775),
-            ),
-        )?;
-        create_directory(
-            config_dir.join("operations"),
-            PermissionEntry::new(
-                Some(self.user.clone()),
-                Some(self.group.clone()),
-                Some(0o775),
-            ),
-        )?;
+                Some(mode),
+            )
+        };
+        create_directory(&config_dir, permissions(0o775))?;
+        create_directory(config_dir.join("mosquitto-conf"), permissions(0o775))?;
+        create_directory(config_dir.join("operations"), permissions(0o775))?;
         create_directory(
             config_dir.join("operations").join("c8y"),
-            PermissionEntry::new(
-                Some(self.user.clone()),
-                Some(self.group.clone()),
-                Some(0o755),
-            ),
-        )?;
-        create_directory(
-            config_dir.join("plugins"),
-            PermissionEntry::new(
-                Some(self.user.clone()),
-                Some(self.group.clone()),
-                Some(0o775),
-            ),
-        )?;
-        create_directory(
-            config_dir.join("sm-plugins"),
-            PermissionEntry::new(
-                Some(self.user.clone()),
-                Some(self.group.clone()),
-                Some(0o755),
-            ),
-        )?;
-        create_directory(
-            config_dir.join("device-certs"),
-            PermissionEntry::new(
-                Some(self.user.clone()),
-                Some(self.group.clone()),
-                Some(0o775),
-            ),
+            permissions(0o755),
         )?;
+        create_directory(config_dir.join("plugins"), permissions(0o775))?;
+        create_directory(config_dir.join("sm-plugins"), permissions(0o755))?;
+        create_directory(config_dir.join("device-certs"), permissions(0o775))?;
 
         let config = self.context.load_config()?;
 
-        create_directory(
-            config.logs.path.clone(),
-            PermissionEntry::new(
-                Some(self.user.clone()),
-                Some(self.group.clone()),
-                Some(0o775),
-            ),
-        )?;
-
-        create_directory(
-            &config.data.path,
-            PermissionEntry::new(
-                Some(self.user.clone()),
-                Some(self.group.clone()),
-                Some(0o775),
-            ),
-        )?;
-
-        create_directory(
-            config_dir.join(".tedge-mapper-c8y"),
-            PermissionEntry::new(
-                Some(self.user.clone()),
-                Some(self.group.clone()),
-                Some(0o775),
-            ),
-        )?;
+        create_directory(&config.logs.path, permissions(0o775))?;
+        create_directory(&config.data.path, permissions(0o775))?;
+        create_directory(config_dir.join(".tedge-mapper-c8y"), permissions(0o775))?;
```

Comment:
None of these changes should change the behaviour from what it was previously, it just felt like a lot of code that could easily be tidied up.

I was however surprised to see that the mode bits vary (mostly `775`, but occassionally `755`), but couldn't find any explanation in the git history as to why that is. Is it intentional that some directories can only be written to by the `tedge` user, and not the `tedge` group, or are they all meant to be the same?